src/plugins/unified_search/public/filter_bar/filter_item/filter_item.tsx

```diff
@@ -400,9 +392,35 @@ export function FilterItem(props: FilterItemProps) {
     );
   }
 
+  const renderComponent = () => {
+    switch (renderedComponent) {
+      case 'menu':
+      default:
+        return <EuiContextMenu initialPanelId={0} panels={getPanels()} />;
+      case 'editFilter':
+        return (
+          <EuiContextMenuPanel
+            items={[
+              <div style={{ width: FILTER_EDITOR_WIDTH, maxWidth: '100%' }}>
+                <FilterEditor
+                  filter={filter}
+                  indexPatterns={indexPatterns}
+                  onSubmit={onSubmit}
+                  onCancel={() => {
+                    setIsPopoverOpen(false);
+                  }}
+                  timeRangeForSuggestionsOverride={props.timeRangeForSuggestionsOverride}
+                />
+              </div>,
+            ]}
+          />
+        );
+    }
+  };
+
   return (
     <EuiPopover anchorPosition="downLeft" {...popoverProps}>
-      <EuiContextMenu initialPanelId={0} panels={getPanels()} />
+      {renderComponent()}
```

Comment:
`swith-case` operator confuse me cause we have only 2 cases. Can we refactor it to: 

```
{
     renderedComponent === editFilter ? <EuiContextMenuPanel
            items={[
              <div style={{ width: FILTER_EDITOR_WIDTH, maxWidth: '100%' }}>
                <FilterEditor
                  filter={filter}
                  indexPatterns={indexPatterns}
                  onSubmit={onSubmit}
                  onCancel={() => {
                    setIsPopoverOpen(false);
                  }}
                  timeRangeForSuggestionsOverride={props.timeRangeForSuggestionsOverride}
                />
              </div>,
            ]}
          /> : <EuiContextMenu initialPanelId={0} panels={getPanels()} />
} 
```

Reply:
Yes, sure. You right. It's look like better with 2 cases. Thanks.